cmd/template_cmd.go

```diff
@@ -343,3 +367,114 @@ func TemplateServersList(c *cli.Context) error {
 	}
 	return nil
 }
+
+// =========== Template helpers =============
+
+// convertFlagParamsToCookbookVersions returns the json representation for the given friendly input format of cookbook versions assignation
+// i.e: "wordpress:0.1.0,nano=2.0.1,1password~>1.3.0"
+func convertFlagParamsToCookbookVersions(c *cli.Context, cbvsIn string) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
+	cookbookVersions := make([]*types.CookbookVersion, 0)
+	for _, cbvIn := range strings.Split(cbvsIn, ",") {
+		values := regexp.MustCompile(`(.*?)(~>|=|>=|<=|>|<|:)(\w.*)`).FindStringSubmatch(cbvIn)
```

Comment:
You are compiling the regex in every iteration of the loop. Intead, the templateCookbookVersionValueRegexp should be a package private variable initialize outside this helper function:
 ``` Go
var templateCookbookVersionValueRegexp regexp.Regexp = regexp.MustCompile(`(.*?)(~>|=|>=|<=|>|<|:)(\w.*)`)

func convertFlagParamsToCookbookVersions(c *cli.Context, cbvsIn string) (map[string]interface{}, error) {
	result := map[string]interface{}{}
	cookbookVersions := make([]*types.CookbookVersion, 0)
	for _, cbvIn := range strings.Split(cbvsIn, ",") {
		values := templateCookbookVersionValueRegexp.FindStringSubmatch(cbvIn)
```

Reply:
Regarding the regexp itself, the name of the cookbook should not be empty, so I'd start with `(.+)`.

Reply:
About (.+) with current regular expression takes a false positive
![imagen](https://user-images.githubusercontent.com/22797797/57224063-75ce1480-7008-11e9-8f8d-c61c7b031b06.png)
instead of 
![imagen](https://user-images.githubusercontent.com/22797797/57224089-88e0e480-7008-11e9-919b-991befd13ed1.png)

I would prefer the current one.

Reply:
Done, updated regular expression but not that change

Reply:
You should then validate that the name is not empty, I don't remember if that's being checked

Reply:
Uhm...,done!.
Uhm..., it seems you were right, finally i fixed the regular expression; and adjusted to parse version up to "m.n.r".
Take a look: https://regex101.com/r/4Lxtp8/8